microcosm_flask/fields.py

```diff
@@ -46,3 +49,48 @@ def _deserialize_by_name(self, value, attr, data):
             return getattr(self.enum, value)
         except AttributeError:
             self.fail('by_name', name=value)
+
+
+class TimestampField(Field):
+    """
+    Timestamp valued field, as either a unix timestamp (default) or isoformat string.
+
+    """
+    EPOCH = datetime(1970, 1, 1)
+
+    def __init__(self, use_isoformat=False, *args, **kwargs):
+        self.use_isoformat = use_isoformat
+        super(TimestampField, self).__init__(*args, **kwargs)
+
+    def _serialize(self, value, attr, obj):
+        """
+        Serialize value as a timestamp, either as a Unix timestamp (in float second) or a UTC isoformat string.
+
+        """
+        if value is None:
+            return None
+
+        if self.use_isoformat:
+            return datetime.utcfromtimestamp(value).isoformat()
+        else:
+            return value
+
+    def _deserialize(self, value, attr, obj):
+        """
+        Deserialize value as a Unix timestamp (in float seconds).
+
+        Handle both numeric and UTC isoformat strings.
+
+        """
+        if value is None:
+            return None
+
+        try:
+            return float(value)
```

Comment:
Historically, C-based systems defined timestamps as seconds since the Jan 1 1970 UTC. People needed a way to extend this sub-second timing. The C world added a struct with both seconds and nanoseconds. Java switched to a long in milliseconds. Python opted for floating point seconds.

Since this is Python, I'm using the floating point convention. I don't think there's a universal answer, but float is nice because it's trivial to case to seconds for systems that need that and is easily encoded as a single value, whether in a database or over the network as JSON.